Chess960, Loading Custom FEN positions, etc...

Far-Aiir opened the pull request:
# New Stuff
Added Chess960 and Custom FEN Position loading. (Edited util.py, db.py, config.py, bot.py)
Added a requirements.txt
Added a reset command so bot admins can reset users by mentioning them. Shouldn't need to use a debug command to reset a user.
# Changes to old code
Allowed the use of users ID's with the games command and also added game length to the display of the games command.

Far-Aiir committed:
Chess960, Loading Custom FEN positions, etc...

Added Chess960 and Custom FEN Position loading. (Edited Util, db, config)
Added a requirements.txt
Allowed the use of users ID's with the games command and also added game length to the display of the games command.

qwertyquerty commented on commands/games.py:
```diff
@@ -41,7 +47,7 @@ class C_Games(Command):
         gs = gs[page*PAGELENGTH:(page+1)*PAGELENGTH]
         for g in gs:
-            em.add_field(name=str(g._id),value="{} vs {} ({})".format(db.User.from_id(g.white).name, db.User.from_id(g.black).name, OUTCOME_NAMES[g.outcome]), inline=False)
+            em.add_field(name="{} ⁠— {} Moves".format(g._id, math.ceil(len(db.Game.from_id(g._id).moves) / 2)), value="{} vs {} ({})".format(db.User.from_id(g.white).name, db.User.from_id(g.black).name, OUTCOME_NAMES[g.outcome]), inline=False)
```
I don't like the moves displaying in the name, should be somewhere in the value

qwertyquerty commented on commands/newgame.py:
```diff
@@ -25,11 +25,30 @@ class C_Newgame(Command):
+						if variant == VARIANT_CUSTOMFEN:
+							try:
+								board = chess.Board()
+								joinedArgs = " ".join(ctx.args)
```
Don't use camelcase anywherw

qwertyquerty commented on commands/reset.py:
```diff
@@ -0,0 +1,20 @@
+    
+    @classmethod
+    async def run(self,ctx):
+        if len(ctx.mentions) > 0:
```
This can just be `if len(ctx.mentions)`

qwertyquerty commented on db.py:
```diff
@@ -69,6 +69,13 @@ def new(cls,u1,u2, variant=VARIANT_STANDARD, fen=None, rated=True):
 			elif variant == VARIANT_HORDE:fen = chess.variant.HordeBoard().fen()
+			elif variant == VARIANT_960:
+				holder = chess.Board(chess960=True)
+				boardNum = random.randint(0,960)
```
No camelcase

qwertyquerty commented on util.py:
```diff
@@ -96,6 +96,8 @@ def get_base_board(g):
     elif g.variant == VARIANT_RACINGKINGS: board = chess.variant.RacingKingsBoard(fen=g.basefen)
     elif g.variant == VARIANT_HORDE: board = chess.variant.HordeBoard(fen=g.basefen)
+    elif g.variant == VARIANT_960: board = chess.Board(fen=g.basefen)
+    elif g.variant == VARIANT_CUSTOMFEN: board = chess.Board(fen=g.basefen)
```
Are you sure you don't need to supply the number here?

qwertyquerty commented on commands/games.py:
```diff
@@ -8,8 +8,14 @@ class C_Games(Command):
+                a = int(ctx.args[0])
+            else:
+                a = int(ctx.mentions[0].id)
+        if db.User.from_id(int(a)).exists:
```
No need to have 2 db operations here when you could have one. Also .exists is not needed. If exists is false the entire object is falsey

qwertyquerty commented on requirements.txt:
```diff
@@ -0,0 +1,6 @@
+tok
```
Tok is not a requirement, thats just the token file you can't see

qwertyquerty commented on commands/newgame.py:
```diff
@@ -25,11 +25,30 @@ class C_Newgame(Command):
+							try:
+								board = chess.Board()
+								joinedArgs = " ".join(ctx.args)
+								joinedArgs = joinedArgs.replace("<@" + str(ctx.mentions[0].id) + ">", "")				
```
You can just do ctx.mentions[0].mention

Far-Aiir committed:
Quick Fixes from last update.

- Replaced camelCase with snake_case
- Changed games.py around
- Removed Tok out of Requirements.txt
- Updated some stuff involving variant code.
